commercial/app/model/commercial/travel/TravelOffersAgent.scala

```diff
@@ -15,7 +15,14 @@ object TravelOffersAgent extends MerchandiseAgent[TravelOffer] with ExecutionCon
   }
 
   def specificTravelOffers(offerIdStrings: Seq[String]): Seq[TravelOffer] = {
-    available filter (offer => offerIdStrings contains offer.id)
+
+    def sortByRequestOrder(a: TravelOffer, b: TravelOffer) = {
+
+      def getOriginalPositionOfId(offer: TravelOffer) = offerIdStrings.indexOf(offer.id)
+      getOriginalPositionOfId(a) < getOriginalPositionOfId(b)
+    }
+
+    available filter (offer => offerIdStrings contains offer.id) sortWith sortByRequestOrder
```

Comment:
Both good suggestions!

In just sorting/augmenting what was there I completely missed the opportunity to simplify, I think I'm going to go for:

`offerIdStrings flatMap (offerId => available find (_.id == offerId))`

It is succinct, keeps the order and doesn't require any sorting.